### PR TITLE
updated PR #1160

### LINKED ISSUE #537: Alert on field content?
I am sure that this has been mentioned somewhere previously but it would be awesome to be able to alert based on field content (string and regex matching) - currently field condition alerts only support metrics & statistics.

Inverting matches should be supported, too.

### PR DESCRIPTION
**Pull request by kroepke**
- this update works against 1.1 to be able to use the fixed tests without merging everything everywhere
 - implement missing tests for FieldStringValueAlertCondition (positive and negative checks)

fixes #537

### CODE REVIEW

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java [hunk 1]
```diff
@@ -116,6 +122,18 @@ public String getDescription() {
         return sb.toString();
     }
 
+    private String buildFieldStringValueDescription() {
```
Comment (joschi) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java]: Useless use of `StringBuilder` in this method. Could be replaced by simple string concatenation which will also make it easier to read.

Comment (kroepke) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java]: agreed

File: graylog2-server/src/main/java/org/graylog2/alerts/types/FieldStringValueAlertCondition.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        this.searches = searches;
+        this.configuration = configuration;
+        this.field = (String) parameters.get("field");
+        this.value = (String) parameters.get("value");
+    }
+
+    @Override
+    protected CheckResult runCheck() {
+        String filter = "streams:" + stream.getId();
+        String query = new StringBuilder()
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/alerts/types/FieldStringValueAlertCondition.java]: Useless use of `StringBuilder` (see comment in `AlertCondition`).

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java [hunk 3]
```diff
@@ -37,7 +37,8 @@
 
     public enum Type {
         MESSAGE_COUNT,
-        FIELD_VALUE
+        FIELD_VALUE,
+        FIELD_STRING_VALUE
```
Comment (joschi) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java]: Should be `FIELD_CONTENT_VALUE` to be consistent with the actual name of the condition.

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java [hunk 4]
```diff
@@ -87,6 +88,8 @@ public String getSummary() {
                 return "Message count condition";
             case FIELD_VALUE:
                 return "Field value condition";
+            case FIELD_STRING_VALUE:
+                return "Field string value condition";
```
Comment (joschi) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/alerts/AlertCondition.java]: "Field content value condition"

File: graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java [hunk 5]
```diff
@@ -36,6 +36,7 @@
     public enum Type {
         MESSAGE_COUNT,
         FIELD_VALUE,
+        FIELD_STRING_VALUE,
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java]: `FIELD_CONTENT_VALUE`

File: graylog2-server/src/main/java/org/graylog2/alerts/types/FieldStringValueAlertCondition.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import org.graylog2.plugin.streams.Stream;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+public class FieldStringValueAlertCondition extends AbstractAlertCondition {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/alerts/types/FieldStringValueAlertCondition.java]: The class should be named `FieldContentValueAlertCondition` for consistency.
